File: greeting_test.go
package workflow_test

import (
	"testing"

	"github.com/stretchr/testify/require"
)

func Test_module_greeting(t *testing.T) {
	t.Parallel()

	mod := new(module)

	tests := []struct {
		name string
		arg  string
		want string
	}{
		{name: "without arg", arg: "", want: "Hello, World!"},
		{name: "with arg", arg: "Joe", want: "Hello, Joe!"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()

			require.Equal(t, tt.want, mod.greeting(tt.arg))
		})
	}
}

File: test/greeting.test.js
import { check } from "k6";
import { greeting } from "k6/x/workflow_test";

export const options = {
  thresholds: {
    checks: ["rate==1"],
  },
};

export default function () {
  check(greeting(""), {
    "greeting()": (str) => str == "Hello, World!",
  });

  check(greeting("Joe"), {
    "greeting('Joe')": (str) => str == "Hello, Joe!",
  });
}
